src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -88,10 +88,10 @@ public bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs
                    e.Option == RuntimeOptions.FullSolutionAnalysis;
         }
 
-        private bool SupportAnalysisKind(DiagnosticAnalyzer analyzer, string language, AnalysisKind kind)
+        private static bool SupportAnalysisKind(HostAnalyzerManager hostAnalyzerManager, DiagnosticAnalyzer analyzer, string language, AnalysisKind kind)
         {
             // compiler diagnostic analyzer always support all kinds
-            if (HostAnalyzerManager.IsCompilerDiagnosticAnalyzer(language, analyzer))
+            if (hostAnalyzerManager.IsCompilerDiagnosticAnalyzer(language, analyzer))
```

Comment:
this seems weird... we had an ownership and logic flow model that we were previously takin, but not it seems to have just become a set of random static methods on disparate types.  I cannot really understand why these fnuctions are on these types.